code/processes/segmentedtickerplant.q

```diff
@@ -26,11 +24,15 @@ if[.stplg.multilog~`custom;
 // functions used by subscribers
 tablelist:{.stpps.t}
 
-// subscribers who want to replay need this info 
-subdetails:{[tabs;instruments]
- `schemalist`logfilelist`rowcounts`date`logdir!(.ps.subscribe\:[tabs;instruments];.stplg.replaylog[tabs];tabs#.stplg `rowcount;(.eodtime `d);`$getenv`KDBTPLOG)
+
+
+// Subscribers who want to replay need this info
+// Modified to include information about filter used
+subdetails:{[tabs;instruments;segid]
+ `schemalist`logfilelist`rowcounts`date`logdir`filter!(.ps.subscribe\:[tabs;instruments];.stplg.replaylog[tabs];tabs#.stplg `rowcount;(.eodtime `d);`$getenv`KDBTPLOG; .stpps.filtermap\:[tabs;segid])
```

Comment:
I dont think  ` .stpps.filtermap\:[tabs;segid]` should need to use each-left. The filtermap function should return a dictionary with all tables requested so no need to iterate over tables here.